### PR TITLE
Reading kubevirt kubeconfig from credentials

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
This PR creates credentials for the Kubevirt cluster to read the Kubeconfig from the secretes instead of reading them from the cluster object directly

**Does this PR introduce a user-facing change?**:
No

```release-note
NONE
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/retitle Reading kubevirt kubeconfig from credentials

This will be the commit title one this PR is merged, the leading number does not really have to be in the title.

**Comment by moadqassem** [comment]
@alvaroaleman Yes sorry about that I opened the PR via IntelliJ and it took the name from the commit message I guess.

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -394,6 +394,8 @@ type GCPCloudSpec struct {
 
 // KubevirtCloudSpec specifies the access data to Kubevirt.
 type KubevirtCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
 	Kubeconfig string `json:"kubeconfig"`
```
Comment (alvaroaleman) [inline api/pkg/crd/kubermatic/v1/cluster.go]: Could you add an `,omitempty` to the json tag so this field is only marshalled if it has a value?

**Review by alvaroaleman — COMMENTED** [review]
One point and lets make sure the kubevirt e2e tests still pass, other than that, LGTM

**Pull request merged**